sql/core/src/main/scala/org/apache/spark/sql/SparkSessionExtensions.scala

```diff
@@ -74,7 +74,7 @@ import org.apache.spark.sql.execution.{ColumnarRule, SparkPlan}
  *     .config("spark.sql.extensions", "org.example.MyExtensions")
  *     .getOrCreate()
  *
- *   class MyExtensions extends Function1[SparkSessionExtensions, Unit] {
+ *   class MyExtensions extends SparkSessionExtensionsProvider {
```

Comment:
I know that this is based on the existing comments. Can we add a new example instead of removing old example please?

Reply:
sgtm